batch/batch.go

```diff
@@ -0,0 +1,263 @@
+	token, err := iam.GetResourceManagementToken(iam.AuthGrantType())
+
+	if err != nil {
+		return batchARM.Account{}, fmt.Errorf("cannot get auth token: %v", err)
+	}
+
+	accountClient := batchARM.NewAccountClient(helpers.SubscriptionID())
+	accountClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	accountClient.AddToUserAgent(helpers.UserAgent())
+
+	res, err := accountClient.Create(ctx, resourceGroupName, accountName, batchARM.AccountCreateParameters{
+		Location: to.StringPtr(location),
+	})
+
+	if err != nil {
+		return batchARM.Account{}, err
+	}
+
+	err = res.WaitForCompletion(ctx, accountClient.Client)
+
+	if err != nil {
+		return batchARM.Account{}, fmt.Errorf("failed waiting for account creation: %v", err)
+	}
+
+	account, err := res.Result(accountClient)
+
+	if err != nil {
+		return batchARM.Account{}, fmt.Errorf("failed retreiving for account: %v", err)
+	}
+
+	return account, nil
+}
+
+// CreateBatchPool creates an Azure Batch compute pool
+func CreateBatchPool(ctx context.Context, accountName, accountLocation, poolID string) error {
+	token, err := iam.GetBatchToken(iam.AuthGrantType())
+
+	if err != nil {
+		return fmt.Errorf("cannot get auth token: %v", err)
+	}
+
+	poolClient := batch.NewPoolClientWithBaseURI(getBatchBaseURL(accountName, accountLocation))
+	poolClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	poolClient.AddToUserAgent(helpers.UserAgent())
+	poolClient.RequestInspector = fixContentTypeInspector()
+
+	toCreate := batch.PoolAddParameter{
+		ID: &poolID,
+		VirtualMachineConfiguration: &batch.VirtualMachineConfiguration{
+			ImageReference: &batch.ImageReference{
+				Publisher: to.StringPtr("Canonical"),
+				Sku:       to.StringPtr("16.04-LTS"),
+				Offer:     to.StringPtr("UbuntuServer"),
+				Version:   to.StringPtr("latest"),
+			},
+			NodeAgentSKUID: to.StringPtr("batch.node.ubuntu 16.04"),
+		},
+		MaxTasksPerNode:      to.Int32Ptr(1),
+		TargetDedicatedNodes: to.Int32Ptr(1),
+		// Create a startup task to run a script on each pool machine
+		StartTask: &batch.StartTask{
+			ResourceFiles: &[]batch.ResourceFile{
+				batch.ResourceFile{
+					BlobSource: to.StringPtr("https://gist.githubusercontent.com/lawrencegripper/795a9e809b52a0b8f874251c62a5a106/raw/8bfe4f4c80440204c994287e415c56f140dfd747/echohello.sh"),
+					FilePath:   to.StringPtr("echohello.sh"),
+					FileMode:   to.StringPtr("777"),
+				},
+			},
+			CommandLine:    to.StringPtr("bash -f echohello.sh"),
+			WaitForSuccess: to.BoolPtr(true),
+			UserIdentity: &batch.UserIdentity{
+				AutoUser: &batch.AutoUserSpecification{
+					ElevationLevel: batch.Admin,
+					Scope:          batch.Task,
+				},
+			},
+		},
+		VMSize: to.StringPtr("standard_a1"),
+	}
+
+	poolCreate, err := poolClient.Add(ctx, toCreate, nil, nil, nil, nil)
+
+	if err != nil {
+		return fmt.Errorf("cannot create pool: %v", err)
+	}
+
+	if poolCreate.StatusCode != 201 {
```

Comment:
Is `200` status code not good?

Reply:
Probably being too pedantic - did it as the REST spec for Batch API says we're going to get a [201](https://docs.microsoft.com/en-us/rest/api/batchservice/pool/add) back but probably better to check for success code

Reply:
Turns out it's all redundant anyway as autorest is returning err is it's not a 200 or 201 so I can remove these checks. https://github.com/Azure/azure-sdk-for-go/blob/master/services/batch/2017-09-01.6.0/batch/pool.go#L155